src/openrct2/ride/Vehicle.cpp

```diff
@@ -8249,6 +8254,8 @@ static bool vehicle_update_track_motion_forwards(rct_vehicle * vehicle, rct_ride
     uint16 trackTotalProgress = vehicle_get_move_info_size(vehicle->var_CD, vehicle->track_type);
     if (regs.ax >= trackTotalProgress)
     {
+        vehicle_update_crossings(vehicle);
```

Comment:
Is this the correct place to put this. Should this not be after a movement to a new track piece? I would also like to see this guarded by a flag so that only transport rides enter this code. There is no point having rollercoasters enter this code.

Reply:
That is a good point, moving it to after moving would mean doing one offset calculation for the reservation fewer. The ordering here is an artefact from me snipping the code out of other things.